### PR TITLE
Adding connection level tests

### PR DESCRIPTION
**Pull request by mikaelm12**
I had to remove the includes and make copies of the definitions of Format, IChannelConnection and ChannelConnection because there were issues with the conflicting namespaces.

### CODE REVIEW

File: src/Microsoft.AspNetCore.Sockets.Client/project.json [hunk 1]
```diff
@@ -20,11 +20,7 @@
     ],
     "xmlDoc": true,
     "compile": {
-      "include": [
-        "../Microsoft.AspNetCore.Sockets/IChannelConnection.cs",
-        "../Microsoft.AspNetCore.Sockets/Internal/ChannelConnection.cs",
-        "../Microsoft.AspNetCore.Sockets/Format.cs"
-      ]
+      "include": [ ]
```
Comment (moozzyk) [inline src/Microsoft.AspNetCore.Sockets.Client/project.json]: remove

File: src/Microsoft.AspNetCore.Sockets.Client/IChannelConnection.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
```
Comment (moozzyk) [inline src/Microsoft.AspNetCore.Sockets.Client/IChannelConnection.cs]: @davidfowl - is this moving to Channels?

File: src/Microsoft.AspNetCore.Sockets.Client/Connection.cs [hunk 3]
```diff
@@ -5,9 +5,10 @@
 using System.Net.Http;
 using System.Threading.Tasks;
 using System.Threading.Tasks.Channels;
-using Microsoft.AspNetCore.Sockets.Internal;
+using Microsoft.AspNetCore.Sockets.Client.Internal;
 using Microsoft.Extensions.Logging;
 
+
```
Comment (moozzyk) [inline src/Microsoft.AspNetCore.Sockets.Client/Connection.cs]: nit: remove extraneous line

File: test/Microsoft.AspNetCore.SignalR.Tests/EndToEndTests.cs [hunk 4]
```diff
@@ -45,5 +49,42 @@ public EndToEndTests(ServerFixture serverFixture)
...
+        }
+
+
+        private static async Task<string> ReceiveMessage(Connection connection)
+        {
+            Message message;
+            while (await connection.Input.WaitToReadAsync())
+            {
+                if (connection.Input.TryRead(out message))
+                {
```
Comment (moozzyk) [inline test/Microsoft.AspNetCore.SignalR.Tests/EndToEndTests.cs]: I think the message should be disposed.

File: test/Microsoft.AspNetCore.SignalR.Tests/EndToEndTests.cs [hunk 5]
```diff
@@ -45,5 +49,42 @@ public EndToEndTests(ServerFixture serverFixture)
...
+                        ReadableBuffer.Create(Encoding.UTF8.GetBytes(message)).Preserve(),
+                        Format.Text));
+
+                    var received = await ReceiveMessage(connection).OrTimeout();
+                    Assert.Equal(message, received);
+                }
+            }
+        }
+
+
```
Comment (moozzyk) [inline test/Microsoft.AspNetCore.SignalR.Tests/EndToEndTests.cs]: nit: extra line

File: test/Microsoft.AspNetCore.SignalR.Tests/EndToEndTests.cs [hunk 6]
```diff
@@ -45,5 +49,42 @@ public EndToEndTests(ServerFixture serverFixture)
                 await ws.CloseAsync(WebSocketCloseStatus.Empty, "", CancellationToken.None);
             }
         }
+
+        [Fact]
+        public async Task CheckConnection()
```
Comment (moozzyk) [inline test/Microsoft.AspNetCore.SignalR.Tests/EndToEndTests.cs]: rename to something more descriptive (e.g. ConnectionCanSendReceiveMessages)

File: src/Microsoft.AspNetCore.Sockets.Client/Format.cs [hunk 7]
```diff
@@ -0,0 +1,11 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Sockets.Client
+{
+    public enum Format
```
Comment (anurse) [inline src/Microsoft.AspNetCore.Sockets.Client/Format.cs]: Hmm. We have a SignalR.Common we use for sharing types like this. Maybe we need a Sockets.Common?

File: src/Microsoft.AspNetCore.Sockets.Client/IChannelConnection.cs (hunk 2 above)
Comment (anurse) [inline src/Microsoft.AspNetCore.Sockets.Client/IChannelConnection.cs]: If we did Sockets.Common we could at least move this there until it is accepted into Channels. We should probably just submit a PR to corefxlab

File: test/Microsoft.AspNetCore.SignalR.Tests/EndToEndTests.cs (hunk 4 above)
Comment (anurse) [inline test/Microsoft.AspNetCore.SignalR.Tests/EndToEndTests.cs]: Indeed it should, you'll need a `using` block inside the `if`. Technically it doesn't need to be disposed in this scenario, but we want to ensure we're using the right pattern for when we introduce pooling (disposing would return message buffers to the pool)

File: src/Microsoft.AspNetCore.Sockets.Client/Connection.cs (hunk 3 above)
Comment (BrennanConroy) [inline src/Microsoft.AspNetCore.Sockets.Client/Connection.cs]: nit

**Comment by mikaelm12** [comment]
I created Sockets.Common. It contains `IChannelConnection`, `ChannelConnection`,`Format`, and as per @davidfowl's suggestion `Message` .

**Comment by davidfowl** [comment]
@mikaelm12 CI is failing

**Comment by mikaelm12** [comment]
Updated

**Pull request merged**